VerteX rebranded (#1376)
spectrocloud/librarium: content/docs/12.5-vertex/02-install-palette-vertex/10-install-on-vmware.md

---

> title: "Install on VMware vSphere"

Reviewer: ```suggestion
title: "Install Palette VerteX on VMware vSphere"
```

> metaTitle: "Install on VMware vSphere"

Reviewer: ```suggestion
metaTitle: "Install Palette VerteX on VMware vSphere"
```

> metaDescription: "Learn how to install Palette on VMware vSphere."

Reviewer: ```suggestion
metaDescription: "Learn how to install Palette VerteX on VMware vSphere."
```

icon: ""
hideToC: false
fullWidth: false
---

import Tabs from 'shared/components/ui/Tabs';
import WarningBox from 'shared/components/WarningBox';
import InfoBox from 'shared/components/InfoBox';

# Install Palette VerteX on VMware vSphere

> You install Palette VerteX using a Command Line Interface (CLI) that guides you for details. The installer creates a three-node enterprise cluster for high availability (HA): 6 vCPU, 12 GB memory, 70 GB storage. You can invoke the installer on any Linux x86-64 system that has the docker daemon installed and connectivity to the Palette Management console or VMware vSphere. 

Reviewer: ```suggestion
You install Palette VerteX using a Command Line Interface (CLI) that guides you for details. The installer creates a three-node enterprise cluster for high availability (HA): 6 vCPU, 12 GB memory, 70 GB storage. You can invoke the installer on any Linux x86-64 system that has the docker daemon installed and connectivity to the Palette Management console and to the VMware vSphere where Palette VerteX should be deployed to. 
```

Questions:
- are you sure that the CLI deploys a 3-nodes cluster? IIRC, it deploys a single node that scale out to 3 nodes after, no?
- "6 vCPU, 12 GB memory, 70 GB storage": is that for each node? or is it the sum of all node resources?

# Prerequisites

- A Palette VerteX account. Refer to [Access Palette VerteX](/vertex#accesspalettevertex).

> - Downloaded PCG installer CLI.

Reviewer: ```suggestion
- Downloaded Palette CLI.
```

- An Ubuntu Pro Subscription. Ubuntu Pro is FIPS 140-2 certified.

- An Ubuntu Pro token.

> - In VMware environments with internet connectivity, you will need added port permissions to inbound rules for security groups to provide Palette VerteX connectivity and outbound connections. Refer to [Network Ports](/architecture/networking-ports/#self-hostednetworkcommunicationsandports) for a list of required ports that must enabled for inbound or outbound communication.

Reviewer: ```suggestion
- In VMware environments with internet connectivity, you will need to configure firewall ports to allow relevant traffic for Palette to function correctly. Refer to [Network Ports](/architecture/networking-ports/#self-hostednetworkcommunicationsandports) for a list of required ports that must enabled for inbound or outbound communication.
```

> - A new or existing SSH key pair to access the PCG installer for any troubleshooting.

Reviewer: ```suggestion
- A new or existing SSH key pair to access the installer for any troubleshooting.
```

<!-- <InfoBox>

> Self-hosted Palette installations provide a system PCG out-of-the-box and typically do not require a separate, user-installed PCG. However, you can create additional PCGs as needed to support provisioning into remote data centers that do not have a direct incoming connection from the management console.

Reviewer: Suggestion to have a link to VMware PCG installation page.

</InfoBox> -->

> # Install the PCG

Reviewer: ```suggestion
# Install the Enterprise Cluster
```

1. Download the Palette installer. Refer to [Download and Setup](/palette-cli/install-palette-cli#downloadandsetup).

2. Invoke the installer by using the following command. The installer prompts you for configuration details and then initiates the installation. For more information about the ``ec`` subcommand, refer to [Palette Commands](/palette-cli/commands#ec). 

```bash
palette ec install
```

3. When prompted to enable Ubuntu Pro, type `y` and provide your Ubuntu Pro token. This enables the Ubuntu license.  

4. When prompted to enable FIPS, type `y`. The repository location is displayed. 

5. Type your Palette username and password.

6. Choose ``VMware vSphere`` as the cloud type.

7. When prompted, enter the information listed in each of the following tables.

<!-- #### Palette PCG Parameters >DOES THIS SECTION APPLY?

|**Parameter**       | **Description**|
|:-----------------------------|---------------|
|**Cloud Type**| Choose ``VMware vSphere``.|
|**Private Cloud Gateway Name** | Enter a custom name for the PCG. Example: ``vmware-vsphere-pcg-1``.|

> |**Share PCG Cloud Account across platform Projects** |Enter 'y' if you want the Cloud Account associated with the PCG to be available from all Projects within your Organization. Enter 'n' if you want the Cloud Account to be available at the tenant admin scope only.| -->

Author: I'll double check it.

|**Parameter**                            | **Description**|
|-----------------------------------------|----------------|
|**vSphere Endpoint** | vSphere endpoint: FQDN or IP address without a scheme - that is, without an IP protocol, such as ``https://``. Example: `vcenter.mycompany.com`.|
|**vSphere Username** | vSphere account username.|
|**vSphere Password**| vSphere account password.|
|**Allow Insecure Connection** | Bypasses x509 verification. Enter `y` if using a vSphere instance with self-signed Transport Layer Security (TLS) certificates. Otherwise, enter `n`.|

<br />

#### VMware vSphere PCG Cluster Configuration

|**Parameter**                            | **Description**|
|-----------------------------------------|----------------|
|**Datacenter**| |
|**Folder** | |
|**Fault Domain(s)** | Configure one or more fault domains by selecting values for these properties: Cluster, Network, Resource Pool, and Storage Type (Datastore or VM Storage Policy). |
|**NTP Server(s)** | You can provide a list of Network Time Protocol (NTP) servers.  |
|**SSH Public Key(s)** | Provide any public SSH keys you will use to access clusters in your project. |

> |**PCG cluster size** | Specify **1** or **3** nodes (HA). |

Author: This might be copy/paste info that doesn't belong. I'll double check. I think it doesn't apply since the installer creates a three-node EC cluster by default.
